bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableIO.java

```diff
@@ -1108,16 +1111,45 @@ public CloudBigtableMultiTableWriteFn(CloudBigtableConfiguration config) {
     @Override
     public void processElement(ProcessContext context) throws Exception {
       KV<String, Iterable<Mutation>> element = context.element();
-      String tableName = element.getKey();
-      try (Table t = getConnection().getTable(TableName.valueOf(tableName))) {
-        List<Mutation> mutations = Lists.newArrayList(element.getValue());
-        int mutationCount = mutations.size();
-        t.batch(mutations, new Object[mutationCount]);
-        mutationsCounter.addValue((long) mutationCount);
+      List<Mutation> mutations = Lists.newArrayList(element.getValue());
```

Comment:
I think it might be better to iterate over the group rather then materializing the whole collection in memory. This will allow the mutator to push back on the incoming stream.